programs/ssl/ssl_client2.c

```diff
@@ -75,8 +75,8 @@ int main( void )
 #include <stdlib.h>
 #include <string.h>
 
-#define MAX_REQUEST_SIZE      20000
-#define MAX_REQUEST_SIZE_STR "20000"
+#define MAX_REQUEST_SIZE      100
```

Comment:
I don't think this is going to work well with `ssl-opt.sh`. We have multiple tests with `request_size=123` (arguably for no good reason, so this could be changed to a lower value), and some with `request_size=$(( MAX_CONTENT_LEN + 1 ))`. However those test are skipped when `MAX_CONTENT_LEN < 4096`, so perhaps here we could define `MAX_REQUEST_SIZE` as either `MAX_CONTENT_LEN + 1` or a lower value depending on the value of `MAX_REQUEST_SIZE` (and leave appropriate comments here and in `ssl-opt.sh`.

Alternatively, we could allow `MAX_REQUEST_SIZE` to be defined on on the compiler's command line, providing only a default here, and defined a low enough value in `baremetal.sh`. Perhaps that would be a bit cleaner than having non-trivial logic tying this file to `ssl-opt.sh`.

Reply:
Oh yes, of course... thank you for spotting. I have reverted this change for now.